### PR TITLE
Add "custom url" attribute field

### PR DESCRIPTION
**Pull request by goldenapples**
"Custom" is available as an option for the "linkto" attribute, but I hadn't included a field where users could actually specify the custom URL to link to. Adding that here.

Also, I'm adding description fields for "caption" and "alt" to warn about the issues including quote marks or HTML markup in these fields.

### CODE REVIEW

File: inc/class-img-shortcode.php [hunk 1]
```diff
@@ -195,7 +203,8 @@ public static function callback( $attr ) {
 		$image_html = apply_filters( 'img_shortcode_output_img_tag', $image_html, $attr );
 
 		// If a link is specified, wrap the image in a link tag
-		if ( ! empty( $attr['linkto'] ) || ! empty( $attr['url'] ) ) {
+		if ( in_array( $attr['linkto'], array( 'file', 'attachment' ) ) ||
+				( 'custom' === $attr['linkto'] && ! empty( $attr['url'] ) ) ) {
```
Comment (danielbachhuber) [inline inc/class-img-shortcode.php]: What if `$attr['linkto']` isn't set?

File: inc/class-img-shortcode.php [hunk 2]
```diff
@@ -99,6 +101,12 @@ public static function get_shortcode_ui_args() {
 						),
 					),
 
+					array(
+						'label'       => esc_attr__( 'Custom link', 'image-shortcake' ),
```
Comment (danielbachhuber) [inline inc/class-img-shortcode.php]: Shouldn't this be `esc_html__` ?

**Comment by danielbachhuber** [comment]
@goldenapples Couple of minor nits

File: inc/class-img-shortcode.php (hunk 2 above)
Comment (goldenapples) [inline inc/class-img-shortcode.php]: Yup, and the other labels as well. Fixed in dffa02f.

File: inc/class-img-shortcode.php (hunk 1 above)
Comment (goldenapples) [inline inc/class-img-shortcode.php]: Should have left that `! empty` check in. Fixed in 36d19ac.

**Comment by goldenapples** [comment]
Fixed to respond to review issues, and merged master. I think this should be ready?

**Pull request merged**